L1Trigger/TrackFindingTracklet/interface/HistImp.h

```diff
@@ -0,0 +1,118 @@
+#ifndef L1Trigger_TrackFindingTracklet_interface_HistImp_h
+#define L1Trigger_TrackFindingTracklet_interface_HistImp_h
+
+#include <TFile.h>
+#include <TH1F.h>
+#include <TEfficiency.h>
+
+#include "L1Trigger/TrackFindingTracklet/interface/HistBase.h"
+
+namespace Trklet {
+
+  class Settings;
+  class Globals;
+
+  class HistImp : public HistBase {
+  public:
+    HistImp();
+
+    virtual ~HistImp() {}
+
+    void init();
+    void close();
+
+    void bookLayerResidual();
+    void bookDiskResidual();
+    void bookTrackletParams();
+
+    virtual void fillTrackletParams(const Settings *settings,
+                                    Globals *globals,
+                                    int seedIndex,
+                                    int iSector,
+                                    double rinv,
+                                    double irinv,
+                                    double phi0,
+                                    double iphi0,
+                                    double eta,
+                                    double ieta,
+                                    double z0,
+                                    double iz0,
+                                    int tp);
+
+    void FillLayerResidual(
+        int layer, int seed, double phiresid, double iphiresid, double zresid, double izresid, bool match);
+
+    virtual void FillDiskResidual(
+        int disk, int seed, double phiresid, double iphiresid, double rresid, double irresid, bool match);
+
+    void bookSeedEff();
+
+    //Efficiency for finding seed
+    virtual void fillSeedEff(int seedIndex, double etaTP, bool eff);
+
+  private:
+    TFile *h_file_;
```

Comment:
we have no time to do that type of migration in time for the PR deadlines, and it is regardless only an internal debugging. we have therefore taking out the HistImp.h/cc from the main code in interface/src (currently moved to a test/ sub-directory). the histogram booking is turned off by default, and if turned on it now does nothing. we thus maintain the function calls so that the functionality can easily be restored for local processing.

Reply:
As a general comment: I don't think that the requirement for filling DQM histograms outside of the code the produces the quantities being monitored is always a good thing, as it requires making event products that could otherwise be transient quantities.

For this specific case filling histograms "inline" may or may not be a good idea.

Either way, I agree with @skinnari that it may be re-engineered after 11.1.0, assuming the current implementation doesn't break anything.